### PR TITLE
docs(contribution): update contribution guideline

### PR DESCRIPTION
**Pull request by donch1989**
Because

- We need to document all the JSON properties in the component definition.

This commit

- Updates contribution guideline.
- Rewraps the content.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -361,33 +385,76 @@ This file defines the component properties:
...
+
+- **`required`** indicates whether the property is required.
+- **`type`**: describes the JSON type of this field, which could be `integer`,
+  `number`, `boolean`, `string`, `array`, or `object`.
+- **`title`** is used by the Console to provide the title of the property.
+- **`description`** is used by Console to show this description in a popup when
+  you click the question mark beside the field.
+- **`instillShortDescription`**: is used by Console to show this under the field
+  (the original description is shown only when hovered). If this value does not
+  exist, it will be the same as the description.
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:407]: Maybe this is a bit too specific? I'm thinking that a slight change in the UX might outdate this documentation. Perhaps we can be a bit more generic:

```suggestion
- **`description`** is used by Console to provide information about this task.
- **`instillShortDescription`**: is a concise version of `description`, used to
  fit smaller spaces such as a component form field. If this value is empty,
  the `description` value will be used.
```

What do you think?

Comment (jvallesm) [reply .github/CONTRIBUTING.md:407]: Also, there's a click vs hover contradiction between both descriptions.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -46,25 +49,38 @@ There are different types of component: **AI**, **Data**, **Application**, **Ope
...
 #### Data
 
-**Data** components play a crucial role in establishing connections with remote data sources, such as IoT devices (e.g., IP cameras), cloud storage services (e.g., GCP Cloud Storage, AWS S3), data warehouses, or vector databases (e.g., Pinecone). These connectors act as the bridge between VDP and various external data sources. Their primary function is to enable seamless data exchange, enhancing Instill VDP's capability to work with diverse data sources effectively. They are defined and initialized in the [data](../data) package.
+**Data** components play a crucial role in establishing connections with remote
+data sources, such as IoT devices (e.g., IP cameras), cloud storage services
+(e.g., GCP Cloud Storage, AWS S3), data warehouses, or vector databases (e.g.,
+Pinecone). These connectors act as the bridge between VDP and various external
+data sources. Their primary function is to enable seamless data exchange,
+enhancing Instill VDP's capability to work with diverse data sources
+effectively. They are defined and initialized in the [data](../data) package.
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:67]: I think the rendered output is a bit cumbersome for these package references, especially the AI package.
CleanShot 2024-06-03 at 11 16 36
CleanShot 2024-06-03 at 11 18 57
```suggestion
effectively. They are defined and initialized in the [`data`](../data) package.
```

**Review by jvallesm — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (donch1989) [reply .github/CONTRIBUTING.md:407]: Agree, removing the behavior in Console is better.

**Review by jvallesm — APPROVED** [review]

**Pull request merged**